amida-tech/indaba-client
# Config update

If this PR fixes a bug, you _must_ add test cases representative of the bug.

Please refer to [Tettra](https://app.tettra.co/teams/amida/pages/amida-pull-request-and-code-review-guide) for PR review guidelines.

#### What does this PR do?
This PR add a settings.js that is loaded by index.html.  This can contain values for the various env variables currently in the project allowing for the end docker container to be configured at runtime rather than compile time.  

The readme and K8s config both imply that env variables can be set at runtime to change these values previously.  this is in error.  the dotenv-webpack module replaces process.env calls at compile time, not runtime, thus not allowing compile time updates.

NOTE: This PR is dependent upon the updates to the Dockerfile from the previous PR.

This change, or an equivalent update using a different method, is needed in order to be able to create automated, reproducible, deployments of the client.

#### Related JIRA tickets:

#### How should this be manually tested?
* Create a settings.js file in an external directory with adjusted values
* From the project directory `docker build -t test .`
* `docker run --rm -p 8087:80 test`
* confirm values being used are NOT the new setting.js
* stop docker container
* `docker run --rm -p 8087:80 -v <path to settings.js>:/usr/share/nginx/html/settings.js test`
* confirm values new values are being used.

#### Background/Context
See "What does this PR do?"
#### Screenshots (if appropriate):
